### ISSUE TITLE
topic_configuration fails to increase replication factor

### ISSUE THREAD

**Issue by mgrubent** [linkedin/cruise-control#862]
# Background
Support for dynamically changing the replication factor of one or more topics was implemented in #590 

# Issue
In a `kafka` cluster where brokers are near the "maximum allowed number of replicas per broker", increasing the replication factor using the `topic_configuration` endpoint can fail with  
```
("Error processing POST request '/topic_configuration' due to: "
 "'com.linkedin.kafka.cruisecontrol.exception.OptimizationFailureException: "
 '[ReplicaCapacityGoal] Replicas in broker 20003 exceeds the maximum allowed '
 "number of replicas per broker: 20000.'.")
```
The replica distribution in the cluster is like the following:  
```
       BrokerState  CpuPct     DiskMB  DiskPct  FollowerNwInRate           Host  LeaderNwInRate  Leaders  NwOutRate  PnwOutRate  Replicas
Broker                                                                                                                                                     
1        ALIVE   27.57 275,289.66     1.44            568.42        somebroker1          620.40     3195   2,038.46    4,064.25      19964
2        ALIVE   28.29 345,545.38     1.81            860.47        somebroker2          546.34     3507   1,935.36    5,030.85      19634
```
# Proposal
**Alternative 1**
The `topic_configuration` endpoint should accept the same `goals` and `skip_hard_goal_check` parameters as does the `rebalance` endpoint.  
This would allow for dynamically skipping goals such as `ReplicaCapacityGoal`  

**Alternative 2**
The `topic_configuration` endpoint should not generate a proposal that violates goals.

**Comment by kidkun** [comment]
Hey @mgrubent 
This support is added recently(versions 2.0.56 and 0.1.59)
https://github.com/linkedin/cruise-control/wiki/Change-topic-replication-factor-though-Cruise-Control
Now `topic_configuration` have almost all the parameters of `rebalance` endpoint.

**Comment by mgrubent** [comment]
Ah, in that case this can be closed in favor of #869
